src/main/kotlin/com/validation/CurrencyCode.kt

```diff
@@ -0,0 +1,11 @@
+package com.validation
+
+import javax.validation.Constraint
+import kotlin.reflect.KClass
+
+@Target(AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.PROPERTY, AnnotationTarget.FIELD)
+@Retention(AnnotationRetention.RUNTIME)
+@Constraint(validatedBy = [CurrencyCodeValidator::class])
+annotation class CurrencyCode(val message: String = "Currently the only supported currency is PLN",
```

Comment:
po co taka adnotacja skoro mógłby to być enum?

Reply:
racja wtedy nie byłoby potrzeby.